src/app/ReadClient.cpp

```diff
@@ -464,5 +521,154 @@ CHIP_ERROR ReadClient::ProcessAttributeDataList(TLV::TLVReader & aAttributeDataL
     ChipLogFunctError(err);
     return err;
 }
+
+CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    CancelLivenessCheckTimer();
+    ChipLogProgress(DataManagement, "Refresh LivenessCheckTime with %d seconds", mFinalSyncIntervalSeconds);
+    err = InteractionModelEngine::GetInstance()->GetExchangeManager()->GetSessionMgr()->SystemLayer()->StartTimer(
+        mFinalSyncIntervalSeconds * kMillisecondsPerSecond, OnLivenessTimeoutCallback, this);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogFunctError(err);
+        ShutdownInternal(err);
+    }
+    return err;
+}
+
+void ReadClient::CancelLivenessCheckTimer()
+{
+    InteractionModelEngine::GetInstance()->GetExchangeManager()->GetSessionMgr()->SystemLayer()->CancelTimer(
+        OnLivenessTimeoutCallback, this);
+}
+
+void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void * apAppState)
+{
+    ReadClient * const client = reinterpret_cast<ReadClient *>(apAppState);
+    ChipLogError(DataManagement, "Subscription Liveness timeout, would shutdown");
+    if (client->IsFree())
+    {
+        ChipLogError(DataManagement,
+                     "ReadClient::OnLivenessTimeoutCallback invoked on a free client! This should be a bug in CHIP stack!");
+        return;
+    }
+    client->ShutdownInternal(CHIP_ERROR_TIMEOUT);
+}
+
+CHIP_ERROR ReadClient::ProcessSubscribeResponse(System::PacketBufferHandle && aPayload)
```

Comment:
I think it's a fair point to switch to a leaner construct here internally within ReadClient. But how much would we actually save? Let's do some back-of-envelope checks...

* `InteractionModelEngine::OnMessageReceived`: Today, the InteractionModelEngine is just a forwarder of the incoming packet and 'routes' it to the right client/handler object to service it. It today, doesn't make assumptions about the subsequent management of that packet, or the lifetime of it after that. If we go down this Span model, it would mean that the IMEngine is explicitly going to toss out that packet buffer after each message has been processed. This is a strong assumption to make, but one I think that is consistent for all IM processing today (i.e no one holds on to the packet buffer for longer than the call to process the packet).

That would save 6 calls to the respective handling objects in this function.

* In most of the handling objects (CommandSender, CommandHandler, ReadClient, etc) that are themselves ExchangeDelegates, they will be provided a PacketBuffer as part of `OnMessageReceived`. In most cases, they just internally position a Reader on that and don't pass around packet buffer handles after that. It's _only_ ReadClient and ReadHandler that do this modal selection between processing read vs. subscribe messages.

So that's two APIs each X 2. So 4.

Total = 10 functions?

Worth it?

The # of call-sites we save would equal the # branches in `InteractionModelEngine::